core/src/main/clojure/xtdb/expression/temporal.clj

```diff
@@ -583,12 +613,21 @@
               :arg-types [:utf8 :utf8 :int :utf8 :int]}]
     (expr/codegen-call expr)))
 
+(defn duration-from-seconds [^double seconds]
+  ;; apparently EE pases durations as long - so here we do as well
```

Comment:
don't need to keep this type of comment, just adds noise to the namespace.

separately, I'd be tempted to have this function handle the different units - e.g. 1000 for millis, 1e6 for micros etc